azurerm/internal/services/containers/container_group_resource.go

```diff
@@ -917,6 +977,32 @@ func expandContainerGroupContainers(d *schema.ResourceData) (*[]containerinstanc
 		containers = append(containers, container)
 	}
 
+	// Determine ports to be exposed on the group level, based on exposed_ports
+	// and on what ports have been exposed on individual containers.
+	if v, ok := d.Get("exposed_port").(*schema.Set); ok && len(v.List()) > 0 {
+		cgpMap := make(map[int32]bool)
+		for _, p := range containerInstancePorts {
+			cgpMap[*p.Port] = true
+		}
+
+		for _, p := range v.List() {
+			portConfig := p.(map[string]interface{})
+			port := int32(portConfig["port"].(int))
+			if !cgpMap[port] {
+				return nil, nil, nil, fmt.Errorf("Port %d is set as an exposed_port on the container group.\n"+
+					"For this port to be exposed on the container group level, it must\n"+
+					"also be exposed on an individual container in the group.", port)
+			}
+			proto := portConfig["protocol"].(string)
+			containerGroupPorts = append(containerGroupPorts, containerinstance.Port{
+				Port:     &port,
+				Protocol: containerinstance.ContainerGroupNetworkProtocol(proto),
+			})
+		}
```

Comment:
I think we'll need to check port:protocol here, not just the port number or we could potentially introduce an error here where one side is TCP and the other UDP?

Reply:
Yes, and the API also responds with a 400 error if the protocols don't match.

I've added a check to see if the protocols match, and also modified the wording of the error message we return.